xil_cloud/docker-compose.yml

```diff
@@ -0,0 +1,139 @@
+version: '3'
+services:
+
+  carlasim:
+    privileged: true
+    image: usdotfhwastol/carma-simulation:carma-simulation-1.0.1
+    networks:
+      carma_sim_net:
+        ipv4_address: 172.2.0.2
+    environment:
+      - DISPLAY=${DISPLAY}
+      - QT_X11_NO_MITSHM=1
+    volumes:
+      - /tmp/.X11-unix:/tmp/.X11-unix
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - capabilities: [gpu]
+    command: "./mosaic.sh -s Town04_20"
+
+  roscore_1:
+    image: usdotfhwastol/carma-base:carma-system-4.2.0
+    networks:
+      carma_net_1:
+        ipv4_address: 172.3.0.2
+    volumes_from:
+      - container:carma-config:ro
+    volumes:
+      - /opt/carma/.ros:/home/carma/.ros
+      - /opt/carma/logs:/opt/carma/logs
+    restart: always
+    environment:
+      - ROS_IP=172.3.0.2
+      - ROS_MASTER_URI=http://172.3.0.2:11311/
+    command: roscore
+
+  platform_1:
+    image: usdotfhwastol/carma-platform:carma-system-4.2.0
+    networks:
+      carma_net_1:
+        ipv4_address: 172.3.0.3
+    container_name: platform_1
+    volumes_from:
+      - container:carma-config:ro
+    volumes:
+      - /opt/carma/logs:/opt/carma/logs
+      - /opt/carma/.ros:/home/carma/.ros
+      - /opt/carma/vehicle/calibration:/opt/carma/vehicle/calibration
+      - /opt/carma/yolo:/opt/carma/yolo
+      - /opt/carma/maps:/opt/carma/maps
+      - /opt/carma/routes:/opt/carma/routes
+      - /opt/carma/data:/opt/carma/data
+      - /opt/carma/simulation:/opt/carma/simulation
+    environment:
+      - ROS_IP=172.3.0.3
+      - ROS_MASTER_URI=http://172.3.0.2:11311/
+    command: bash -c 'wait-for-it.sh 172.3.0.2:11311 -- roslaunch /opt/carma/vehicle/config/carma_docker.launch'
+
+  platform_ros2:
```

Comment:
The naming of these is a bit confusing, platform_1 for "CARMA Platform instance one of N" but then "platform_ros2" with no mention of which platform instance it's associated with. I'd either drop the "_1" from the first service name (so just platform and platform_ros2) or add the _1_ to the second (so platform_1 and platform_ros2_1).

This reminds me that since our docker compose file changes a bit with the ros2 port, we'll need to make some corresponding changes to the docker-compose generator.